node/src/chain.rs

```diff
@@ -168,10 +169,18 @@ pub fn create_firehose_networks(
                     "Configuring firehose endpoint";
                     "provider" => &provider.label,
                 );
+                let subgraph_limit = firehose.limit_for(&config.node);
 
                 let parsed_networks = networks_by_kind
                     .entry(chain.protocol)
                     .or_insert_with(|| FirehoseNetworks::new());
+
+                // Create n FirehoseEndpoints where n is the size of the pool. If a
+                // subgraph limit is defined for this endpoint then each endpoint
+                // instance will have their own subgraph limit.
+                // eg: pool_size = 3 and sg_limit 2 will result in 3 separate instances
+                // of FirehoseEndpoint and each of those instance can be used in 2 different
+                // SubgraphInstances.
```

Comment:
This is more a usability concern, but it might be surprising to users that in this scenario they can have 6 subgraphs using the provider. An alternative would be to make the limit for each pool `subgraph_limit / conn_pool_size` so that the number of subgraphs using the provider would never exceed `subgraph_limit`

Reply:
I will open a separate issue to look into this. Right now we don't have all the validations so this needs some more careful implementations. Situations like conn_pool_size > subgraph_limit need to be validated and rejected. Since this is a blocker for the EVM Executor tests I will come back to it later.